gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/auth/ExternalAuthenticationResource.java

```diff
@@ -0,0 +1,190 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.management.rest.resource.auth;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier;
+import io.gravitee.rest.api.idp.api.authentication.UserDetails;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.*;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.exceptions.InvalidAlgorithmException;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.exceptions.InvalidTokenException;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.ECJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.MACJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.RSAJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.RemoteJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.resolver.PublicKeyResolver;
+import io.gravitee.rest.api.model.UserEntity;
+import io.gravitee.rest.api.security.utils.AuthoritiesProvider;
+import io.gravitee.rest.api.service.common.GraviteeContext;
+import io.gravitee.rest.api.service.exceptions.UserNotFoundException;
+import jakarta.annotation.PostConstruct;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+@Singleton
+@Path("/auth/external")
+public class ExternalAuthenticationResource extends AbstractAuthenticationResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExternalAuthenticationResource.class);
+    protected static final String ORG_CLAIM = "org";
+    protected static final String ENVIRONMENT_CLAIM = "env";
+    private boolean enabled;
+    private AbstractKeyProcessor<?> keyProcessor;
+    private Algorithm algorithm;
+
+    @Autowired
+    private AuthoritiesProvider authoritiesProvider;
+
+    private String uiURL;
+
+    @PostConstruct
+    public void afterPropertiesSet() {
+        enabled = environment.getProperty("auth.external.enabled", Boolean.class, false);
```

Comment:
Cannot you use @Value instead?

Reply:
Seems to be not working 
    @Value("${auth.external.enabled:false}")
    private boolean enabled;

    @Value("${auth.external.algorithm:#{null}}")
    private Algorithm algorithm;

    @Value("${auth.external.verificationKey:#{null}}")
    private String verificationKey;

    @Value("${auth.external.issuer:#{null}}")
    private String issuer;

Reply:
Do you want me to investigate the issue or you are happy as it is?

Reply:
No need it is fine